client/styles/ui-components/Button.tsx

```diff
@@ -0,0 +1,90 @@
+import styled from "styled-components";
+
+export type ButtonProps = {
+  children?: React.ReactNode;
+  styling?: string;
+};
+
+export type ButtonStyle = {
+  [key: string]: ButtonStyleProperties;
+};
+
+export type ButtonStyleProperties = {
+  backgroundColor: string;
+  hoverBgColor: string;
+  color: string;
+  border?: string;
+};
+
+export default function Button({ children, styling }: ButtonProps) {
+  const STYLES: ButtonStyle = {
+    normal: {
+      backgroundColor: "#EDF2F7",
+      hoverBgColor: "#E2E9F0",
+      color: "black",
+    },
+    skill: {
+      backgroundColor: "#EDF2F7",
+      hoverBgColor: "#CDCDCD",
+      color: "black",
+    },
+    option: {
+      backgroundColor: "#3182CE",
+      hoverBgColor: "#2B6CB0",
+      color: "white",
+    },
+    positive: {
+      backgroundColor: "#38A169",
+      hoverBgColor: "#2F855A",
+      color: "white",
+    },
+    negative: {
+      backgroundColor: "#DD6B20",
+      hoverBgColor: "#C05621",
+      color: "white",
+    },
+  };
+
+  const styles = STYLES[styling];
+
+  if (!styles) {
+    throw new Error("You did not declare styling props...");
+  }
+
+  return (
+    <Button1
+      style={{
+        "--color": styles.color,
+        "--backgroundColor": styles.backgroundColor,
+        "--hoverBgColor": styles.hoverBgColor,
+        "--border": styles.border,
+      }}
```

Comment:
```suggestion
      style={{
        "--color": styles.color,
        "--backgroundColor": styles.backgroundColor,
        "--hoverBgColor": styles.hoverBgColor,
        "--border": styles.border,
      } as React.CSSProperties}
```

^ should resolve the current type error

Reply:
Done